src/main/java/com/mythsoftware/rest/resource/UserResource.java

```diff
@@ -0,0 +1,40 @@
+package com.mythsoftware.rest.resource;
+
+import com.mythsoftware.rest.model.User;
+import com.mythsoftware.rest.service.UserService;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+/**
+ */
+@Named
+@Path("/user")
+public class UserResource {
+
+    @Inject
+    private UserService userService;
+
+    @GET
+    @Path("/getUsers")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getUsers() {
+        List<User> users = userService.getUsers();
+        return Response.ok(users).build();
+    }
+
+    @GET
+    @Path("/getUserById/{id}")
```

Comment:
Normally it's plural.  I guess it's because it's more intuitive??  For example, you might do something like this to get the user's profile pic:  GET /users/{id}/pic... as opposed to GET /users/{id}/friends.  If it's plural, you can expect a collection.  If it's singular, you can expect a single object.  As far as exposing the data model - it does to a certain extent.. at least the "public" data model.  Rest is very CRUDish and the thinking is that it makes it easier to consume - you know exactly where to go to find documentation on how to do something.  For example.. if you want to add a friend or delete a friend, or accept a friend, traditionally we might have a managed bean with methods like "addFriend", "acceptFriend", etc. but if you want to scale this so a bunch of people can use it, it gets unwieldy if the app gets really big.  With a rest interface, I just need to know one endpoint - /users/{id}/friends.. logically I would use POST to add one, DELETE (with an id) to delete one, and PATCH to accept it.  I would look up the "resource" and see what params I need to send in order to accept it - perhaps {accepted:true}.
Here's a good blog about REST:  http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api